### ISSUE TITLE
Apache error

### ISSUE THREAD

**Issue by than-sad** [bhadaway/stop-spammers#110]
Error You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'WHERE active = 0 AND user_login = 'janessaxxxx' ORDER BY signup_id DESC LIMIT' at line 1 de la base de datos de WordPress para la consulta SELECT * FROM  WHERE active = 0 AND user_login = 'janessaxxxx' ORDER BY signup_id DESC LIMIT 0, 1 realizada por require('wp-load.php'), require_once('wp-config.php'), require_once('wp-settings.php'), do_action('init'), WP_Hook->do_action, WP_Hook->apply_filters, ss_init, ss_check_white, be_load, ss_check_white->process, be_load, chkadminlog->process, wp_authenticate, apply_filters('authenticate'), WP_Hook->apply_filters, bp_core_signup_disable_inactive, BP_Signup::get, referer: https://www.xxxxx.es/wp-login.php

**Comment by bhadaway** [comment]
What version of Apache are you running?

**Comment by than-sad** [comment]
7.3.1

**Comment by than-sad** [comment]
I tried with php 7.3.8 and 7.3.10 with same result. This error appears every time user try to login with non existing or incorrect username and password.

**Comment by bhadaway** [comment]
Ah, that's a PHP error and that's your PHP version, not Apache.

I haven't been able to recreate this error yet.

What version of WordPress, Stop Spammers, and MySQL are you on? Are you able to update those?

**Comment by than-sad** [comment]
Yes sorry, it’s a Apache error log message, not Apache error. WP 5.2.4, SS 2019.1, MSQL 8.0.17. I have to check if there is an update for MSQL.

**Comment by than-sad** [comment]
I see there is a MySQL version 8.0.20. Should I try it?

**Comment by bhadaway** [comment]
No, everything is good on your end. I'm pretty sure the issue is MySQL 8, which Stop Spammer's code has not been tested or updated for.

I'll try and dig into this soon, but I've also put up the "Help Wanted" sign, in case anyone in the community can assist.

Thanks

**Comment by than-sad** [comment]
Thanks. Anyway isn’t a critical error, just annoying to see repeated messages always in error log.

Few weeks ago I found information about this error, but now I can’t locate it. I will try again and post link here to give more information about this error.
Best regards!

**Comment by StrothiP** [comment]
Hey guys,

I think I'm seeing the same issue on my server:

WordPress database error You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ‘WHERE active = 0 AND user_login = ‘admin’ ORDER BY signup_id DESC LIMIT 0, 1′ at line 1 for query SELECT * FROM WHERE active = 0 AND user_login = ‘admin’ ORDER BY signup_id DESC LIMIT 0, 1 made by require(‘wp-load.php’), require_once(‘wp-config.php’), require_once(‘wp-settings.php’), do_action(‘init’), WP_Hook->do_action, WP_Hook->apply_filters, ss_init, ss_check_white, be_load, ss_check_white->process, be_load, chkadminlog->process, wp_authenticate, apply_filters(‘authenticate’), WP_Hook->apply_filters, bp_core_signup_disable_inactive, BP_Signup::get

@bhadaway Did you have a chance to look into what might be causing this?

Kind regards, Philip

**Comment by bhadaway** [comment]
Yes I did.

Fortunately, it doesn't affect the core plugin in any way. This issue is isolated to the beta features, database cleanup, and the vulnerability scan.

And luckily, it's just a simple matter of updating the syntax in the code to be MySQL 8 compatible. This fix will probably happen in the near future.

**Comment by than-sad** [comment]
Hi, after 5 months my apache error log is full of this error. Nothing changed in all new releases...

Are you going to fix it or not?

Regards.

**Comment by than-sad** [comment]
It´s ok, leave it. I found a better plugin.

**Comment by bhadaway** [comment]
Sorry, I was no longer involved with the project for a while, but I'm now working on it again.

_Investigating..._

**Comment by than-sad** [comment]
One year and 10 months after I reported the error is obviously too late.

Im happily using Zero-Spam with no errors and no more spam registrations in buddy press from months ago.

Best regards.

**Comment by bhadaway** [comment]
Always sorry to lose a user, but I understand. It was just me for a few years, and now there's only two of us. We really don't have the kind of resources needed to push updates and fixes at the rate of some of the bigger plugins that make a lot of money with full dev teams, but we're trying to bring in revenue now, and we'll keep at it. Better later than never, for those still using Stop Spammers.

**Comment by bhadaway** [comment]
I'm pretty sure that I've isolated this to https://github.com/Trumani/stop-spammers/blob/master/settings/ss_diagnostics.php. I've checked the SQL syntax throughout the plugin and am only getting syntax errors for:

https://github.com/Trumani/stop-spammers/blob/7dd87291df2dfd948d44d15cace9d2062ef68be8/settings/ss_diagnostics.php#L1117

and

https://github.com/Trumani/stop-spammers/blob/7dd87291df2dfd948d44d15cace9d2062ef68be8/settings/ss_diagnostics.php#L1119

**Comment by bhadaway** [comment]
The original author of Stop Spammers had simply disabled the vulnerability scanner altogether because it was incomplete and beta at best.

It's really hard, basically impossible to know what features are used and not used by Stop Spammers users, because we'd never be able to get enough feedback.

It's possible we might just need to remove the feature altogether, if it's too problematic.
